git-link.el

```diff
@@ -233,6 +236,8 @@ As an example, \"gitlab\" will match with both \"gitlab.com\" and
 (defun git-link--branch ()
   (or (git-link--get-config "git-link.branch")
       git-link-default-branch
+      (when (git-link--using-magit-blob-mode)
+        (magit-rev-branch magit-buffer-revision))
```

Comment:
Under what magi command does this return a valid branch? I tried with a file opened from a commit via `magit-log` but no branch.

Reply:
I open a blob from a branch using `magit-find-file`, and then `(magit-rev-branch magit-buffer-revision)` evaluates to the name of that branch.

You likely opened a blob that is not from the tip of a branch, in which case `magit-buffer-revision` will have the right commit id, but `(magit-rev-branch magit-buffer-revision)` will evaluate to nil.  In this case, I believe the right thing to do would be to return the commit id, but I see that `git-link` is in fact using the checked-out branch.  I will amend my changes to fix this problem.